sc_analysis_dump, sc_tracediff, sc_ttlexp, sc_warts2csv, sc_warts2json, sc_warts2text, sc_wartscat, sc_wartsdump, sc_wartsfilter, sc_wartsfix, scamper: add page (#12230)
tldr-pages/tldr: pages/common/scamper.md

# scamper

> A tool that actively probes the Internet in order to analyze topology and performance.

> > More information: <https://www.caida.org/catalog/software/scamper/>.

Author: Maybe something like this?
```
> Includes some tools that start with `sc_`, for example `sc_warts2text` or `sc_ttlexp`.
```

# scamper

> > A tool that actively probes the Internet in order to analyze topology and performance.

Reviewer: ```suggestion
> Actively probe the Internet in order to analyze topology and performance.
```

Author: > Like its predecessor [skitter](https://www.caida.org/catalog/software/skitter/), scamper is a feature-rich tool that actively probes the Internet in order to analyze topology and performance.

Here I have "stolen" the text from the [website](https://www.caida.org/catalog/software/scamper/).

Reviewer: Yeah, I realized it, and that's OK, but it's recommended that our descriptions follow the style guide. "A tool that" is kinda useless, many commands could add that to the beggining of the description and that wouldn't add any value. We have some specific rules in our [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#imperative-mood) that makes all descriptions standardized:
> ### Imperative Mood
> 
> - **All descriptions must be concise and phrased in the imperative mood.**
> - This also applies to all translations by default unless otherwise specified in the language-specific section below.
> - For example, when writing documentation for `cd`, a tool to check out and work on a specific directory in the Terminal or Command Prompt, **do not** write a lengthy description such as:
> 
> ```md
> > `cd` is a system tool, available in Windows, macOS, and Linux, to check out a specific directory to get things done in the Command Prompt, Terminal, and PowerShell.
> ```
> 
> It should instead be simplified to make it easier for everyone to read:
> 
> ```md
> > Change the current working directory.
> ```

> Includes some tools that start with `sc_`, for example `sc_warts2text` or `sc_ttlexp`.
> More information: <https://www.caida.org/catalog/software/scamper/>.

> - Execute the standard option (traceroute) to a destination:

Reviewer 2: ```suggestion
- Execute the standard option (`traceroute`) to a destination:
```

`scamper -i {{192.0.2.1}}`

> - Execute two actions (ping and traceroute) on two different targets:

Reviewer 2: ```suggestion
- Execute two actions (`ping` and `traceroute`) on two different targets:
```

`scamper -I "{{ping}} {{192.0.2.1}}" -I "{{trace}} {{192.0.2.2}}`

- Ping several hosts with UDP, use a specific port number for the first ping and increase it for each subsequent ping:

`scamper -c "{{ping}} -P {{UDP-dport}} -d {{33434}}" -i {{192.0.2.1}} -i {{192.0.2.2}}`

> - Use the Multipath Discovery Algorithm (MDA) to determine the presence of load-balanced paths to the destination and use ICMP echo packets to sample with a maximum of three attempts, write the result to a warts file:
> 
> `scamper -O {{warts}} -o {{path/to/output.warts}} -I "{{tracelb}} -P {{ICMP-echo}} -q {{3}} {{192.0.2.1}}"`
> 
> - Execute a Paris traceroute with ICMP to a destination and saves the result in a compressed warts file:
> 
> `scamper -O {{warts.gz}} -o {{path/to/output.warts}} -I "{{trace}} -P {{icmp-paris}} {{2001:db8:dead:beaf::4}}"`
> 
> - Record all ICMP packets that arrive at a specific IP address and have a specific ICMP ID in a warts file:

Reviewer 2: ```suggestion
- Use the Multipath Discovery Algorithm (MDA) to determine the presence of load-balanced paths to the destination and use ICMP echo packets to sample with a maximum of three attempts, write the result to a `warts` file:

`scamper -O {{warts}} -o {{path/to/output.warts}} -I "{{tracelb}} -P {{ICMP-echo}} -q {{3}} {{192.0.2.1}}"`

- Execute a Paris traceroute with ICMP to a destination and save the result in a compressed `warts` file:

`scamper -O {{warts.gz}} -o {{path/to/output.warts}} -I "{{trace}} -P {{icmp-paris}} {{2001:db8:dead:beaf::4}}"`

- Record all ICMP packets that arrive at a specific IP address and have a specific ICMP ID in a `warts` file:
```
